_rules/text-contrast-afw4f7.md

```diff
@@ -0,0 +1,376 @@
+<script>
+	const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+	shadowRoot.textContent = '<span style="color: #333;">Some text in English</span>'
+</script>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This light gray text has a contrast ratio of 2.3:1 on the white background.
+
+```html
+<p style="color: #AAA; background: white;">
+	Some text in English
+</p>
+```
+
+#### Failed Example 2
+
+This light gray text has a contrast ratio between 1.2:1 and 2.3:1 on the white to blue gradient background.
+
+```html
+<p style="color: #AAA; background: linear-gradient(to right, #FFF, #00F); width: 300px">
+	Some text in English
+</p>
+```
+
+#### Failed Example 3
+
+This light gray text has a contrast ratio between 2.7:1 and 3:1 on the background image.
+
+```html
+<p
+	style="color: #555; height:50px; padding-top:20px; background: black no-repeat -20px -20px url('../test-assets/contrast/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Failed Example 4
+
+This black text with 30% alpha channel has a contrast ratio of 2.1:1 on the white background.
+
+```html
+<p style="color: rgba(0,0,0,.3); background: #FFF">
+	Some text in English
+</p>
+```
+
+#### Failed Example 5
+
+This black text with 30% opacity has a contrast ratio of 2.1:1 on the white background.
+
+```html
+<div style="background: #FFF">
+	<p style="color: #000; opacity: .3">
+		Some text in English
+	</p>
+</div>
+```
+
+#### Failed Example 6
+
+This light gray text has a contrast ratio of 2.3:1 on the white background in a shadow DOM tree.
+
+```html
+<p style="color: #aaa; background: #fff;" id="p"></p>
+<script>
+	const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+	shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+#### Failed Example 7
+
+This semi-transparent gray text has a contrast ratio between 2.3:1 and 4.2:1 on the black and white background. The light gray text is compared to the white section of the background and the dark gray text is compared to the black section of the background.
+
+```html
+<style>
+	#backgroundSplit {
+		color: rgba(90, 90, 90, 0.8);
+		background-position: top 0 left 0;
+		background-image: linear-gradient(90deg, transparent, transparent 3.3em, black 3.3em, black 6em);
+		padding: 0 1em;
+	}
+</style>
+<span id="backgroundSplit">
+	Hello world
+</span>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+This text is not [visible][] because of `display: none`.
+
+```html
+<p style="display: none">Some invisible text in English</p>
+```
+
+#### Inapplicable Example 2
+
+This text is not [visible][] because it is positioned off screen.
+
+```html
+<p style="position:absolute; top: -999em">Some invisible text in English</p>
+```
+
+#### Inapplicable Example 3
+
+This text is not [visible][] because the foreground color is the same as the background color.
+
+```html
+<p style="color: white; background: white;" aria-hidden="true">Page last updated on 2020/01/10</p>
```

Comment:
```suggestion
<p style="color: white; background: white;" aria-hidden="true">Hidden text - U U D D L R L R B A S</p>
```

Reply:
Page last updated is often useful information that should be visible. I've suggested a fun easter egg, that makes sense to be hidden in a findable way.